factory-package-news/factory-package-news.py

```diff
@@ -89,9 +98,28 @@ def _getdata(h):
                 data['changelogtime'] = h['changelogtime']
                 data['changelogtext'] = h['changelogtext']
                 for (t, txt) in enumerate(data['changelogtext']):
-                    data['changelogtext'][t] = str(txt, 'utf-8')
+                    data['changelogtext'][t] = forcestr(txt)
                 changelogs[srpm] = data
 
+        def _walk_through_iso_image(iso, path="/"):
+            file_stats = iso.readdir(path)
+            if file_stats is None:
+                return None
```

Comment:
Thanks for the review. Indeed it could have happened before as the directory might not exist but now is indeed not the case as we start from / anyway. Will add an exception.